packages/@aws-cdk/aws-scheduler-targets-alpha/lib/universal.ts

```diff
@@ -0,0 +1,134 @@
+import { IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
+import { Aws, Token } from 'aws-cdk-lib';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { awsSdkToIamAction } from 'aws-cdk-lib/custom-resources/lib/helpers-internal';
+import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
+
+/**
+ * AWS read-only API action name prefixes that are not supported by EventBridge Scheduler.
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-targets-universal.html
+ */
+const NOT_SUPPORTED_ACTION_PREFIX = [
+  'get',
+  'describe',
+  'list',
+  'poll',
+  'receive',
+  'search',
+  'scan',
+  'query',
+  'select',
+  'read',
+  'lookup',
+  'discover',
+  'validate',
+  'batchGet',
+  'batchDescribe',
+  'batchRead',
+  'transactGet',
+  'adminGet',
+  'adminList',
+  'testMigration',
+  'retrieve',
+  'testConnection',
+  'translateDocument',
+  'isAuthorized',
+  'invokeModel',
+];
+
+/**
+ * Properties for a Universal Target
+ */
+export interface UniversalProps extends ScheduleTargetBaseProps {
+  /**
+   * The AWS service to call.
+   *
+   * This must be in lowercase.
+   */
+  readonly service: string;
+
+  /**
+   * The API action to call.
+   *
+   * You cannot use read-only API actions such as common GET operations.
+   *
+   * Also, This must be in camelCase.
+   */
+  readonly action: string;
+
+  /**
+   * The resources for the IAM policy statement that will be added to the scheduler role's policy
+   * to allow the scheduler to make the API call.
+   *
+   * We recommend specifying the resources to the minimum required for the API call.
+   *
+   * @default ['*']
+   */
+  readonly iamResources?: string[];
```

Comment:
The props regarding the IAM Policy to be granted to the Role of the Scheduler are based on step-functions-tasks.
I am wondering how to design this area to be user-friendly, so if you have any opinions, please let me know.
https://github.com/aws/aws-cdk/blob/v2.173.3/packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts

Reply:
Making it an optional property means that the default wildcard `['*']` could be specified by forgetting to specify the property. Instead, I prefer the current code, which is more secure by requiring an explicit specification.

```ts
  /**
   * The resources for the IAM policy statement that will be added to the scheduler role's policy
   * to allow the scheduler to make the API call.
   */
  readonly iamResources: string[];
```